### PR TITLE
chore(V3template): m365-tab-js v3 template

### PR DESCRIPTION
**Pull request by frankqianms**
local-debug（in Teams） pass
provision&deploy pass

### CODE REVIEW

File: templates/scenarios/js/m365-tab/teamsfx/app.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    with:
+      storageResourceId: ${{TAB_AZURE_STORAGE_RESOURCE_ID}}
+      indexPage: index.html
+      errorPage: error.html
+    # Output: N/A
+
+deploy:
+  - uses: npm/command # Run npm command
+    with:
+      workingDirectory: .
```
Comment (Siglud) [inline templates/scenarios/js/m365-tab/teamsfx/app.yml:25]: This line can be removed if the dir is "."

File: templates/scenarios/js/m365-tab/teamsfx/app.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  - uses: npm/command # Run npm command
+    env:
+      REACT_APP_CLIENT_ID: ${{AAD_APP_CLIENT_ID}}
+      REACT_APP_START_LOGIN_PAGE_URL: ${{TAB_ENDPOINT}}/auth-start.html
+    with:
+      workingDirectory: .
+      args: run build
+  - uses: azureStorage/deploy # Deploy bits to Azure Storage Static Website
+    with:
+      workingDirectory: .
```
Comment (Siglud) [inline templates/scenarios/js/m365-tab/teamsfx/app.yml:36]: Also can be removed

**Comment by frankqianms** [comment]
> Just reminder, for local debug, still needs to update `tasks.json`, `launch.json` and `package.json`.

u'r right, edited as "local debug(only in Teams) pass"

File: templates/scenarios/js/m365-tab/teamsfx/app.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      distributionPath: ./build # Deploy base folder
+      ignoreFile: ./.appserviceIgnore # Can be changed to any ignore file location, leave blank will ignore nothing
```
Comment (Siglud) [inline templates/scenarios/js/m365-tab/teamsfx/app.yml:38]: Tab projects that use React do not need the ingore file

Comment (frankqianms) [reply templates/scenarios/js/m365-tab/teamsfx/app.yml]: removed

File: templates/scenarios/js/m365-tab/teamsfx/app.yml (hunk 2 above)
Comment (frankqianms) [reply templates/scenarios/js/m365-tab/teamsfx/app.yml]: removed

File: templates/scenarios/js/m365-tab/teamsfx/app.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  - uses: npm/command # Run npm command
+    with:
+      workingDirectory: .
+      args: install
+  - uses: npm/command # Run npm command
+    env:
+      REACT_APP_CLIENT_ID: ${{AAD_APP_CLIENT_ID}}
+      REACT_APP_START_LOGIN_PAGE_URL: ${{TAB_ENDPOINT}}/auth-start.html
+    with:
+      workingDirectory: .
```
[the remark this answers is not in the document]
Comment (frankqianms) [reply templates/scenarios/js/m365-tab/teamsfx/app.yml]: removed

File: templates/scenarios/js/m365-tab/teamsfx/app.yml (hunk 1 above)
Comment (frankqianms) [reply templates/scenarios/js/m365-tab/teamsfx/app.yml]: removed

**Review by Siglud — APPROVED** [review]

**Pull request merged**